azurerm/internal/services/containers/resource_arm_kubernetes_cluster.go

```diff
@@ -1230,6 +1238,10 @@ func expandKubernetesClusterAgentPoolProfiles(input []interface{}, isNewResource
 			return nil, fmt.Errorf("Can't create an AKS cluster with autoscaling enabled but not setting min_count or max_count")
 		}
 
+		if nodeLabels := utils.ExpandMapStringPtrString(config["node_labels"].(map[string]interface{})); len(nodeLabels) > 0 {
```

Comment:
Do we want to empty node label struct to the api so people can remove the labels?

Reply:
From my tests of Azure API node labels can not be changed.
Whenever you try to alter node labels API accepts the update, but labels are never changed.